tests/src/Loader/AssemblyDependencyResolver/AssemblyDependencyResolverTests/AssemblyDependencyResolverTests.cs

```diff
@@ -130,6 +131,85 @@ public void TestAssembly()
             }
         }
 
+        public void TestAssemblyWithCaseDifferent()
+        {
+            // Testing case sensitive file name resolution in Windows
+            // Host policy returns 2 file paths with the casing changed,
+            // AssemblyDependencyResolver should not throw
+            if(RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Shouldn't we at least check that we can successfully create the `AssemblyDependencyResolver` and call `ResolveAssemblyToPath` on all platforms instead of just Windows? The behaviour difference due the OS-specific treatment of casing should just be at whether or not `ResolveAssemblyToPath` returns a path, right?

Reply:
Yes, the source code fix is simply being tolerant on the same simple file name (case different) on the managed side for TPAs. Removed the windows specific check on the test